### DISCUSSION TITLE
Errors in Producing Output

### DISCUSSION THREAD

**Discussion by hanasakura7** [apache/netbeans#7350] [Q&A]
Hello everyone, I need help. I am basically a newbie in CS, currently using NetBeans IDE 19. I was basically just practising my coding from this video: 

https://www.youtube.com/watch?v=Af3s3KsxStY&t=2069s

on the minute of 34:24, this should be the output: 

![Screenshot (274)](https://github.com/apache/netbeans/assets/140406290/c4ec38b8-3989-4579-825b-908191353222)

I got this for my output instead: 
![Screenshot (275)](https://github.com/apache/netbeans/assets/140406290/e2efe343-d24e-4edd-835e-79defb0155f9)

I think I accidentally pressed something because there were errors, but I ended up messing up but I'm not sure what it is. Can someone help me to figure out where the issue lies on? I've reset the Windows (there's an option in Netbeans) but the outcome is still the same. Please help a lost student out here. Thank you.

#java #7342

**Comment by JoachimRohde** [comment]
Code from the video:

`System.out.println(u == u2);`

Your code:

```
System.out.println(u);
System.out.println(u2);
```

The first prints out the result of a reference comparison between two objects. (A comparison with == does not compare the values of the objects rather the references of the objects).
The second prints out whatever the toString()-method of the object returns. (In User.java you will have a toString method overriden which return the "name" and "membership").

So you get a different output because your code is different.

A friendly heads-up for the future: posting (exclusively) screenshots of code is frowned upon (because others cannot copy it to try it out). And to include unnecessary information like the link to ticket 7342 which has nothing to do with your problem.

**Reply by hanasakura7** [reply]
Hi, thank you so much for your response. I'm still a newbie in this so I might have missed something out due to my short-sightedness. And thank you for telling my mistake, I'll continue to improve myself better. Have a nice day :)
